ImgSmall.java

```diff
@@ -59,7 +68,16 @@ public static void resize(String[] args) throws IOException{
 
     File out =
         new File(filename + "_" + args[1] + extension);
-    //Write output to file, using the "jpg" file type
+    //Write output to file
+    if (!extension.equals("png")&&!extension.equals("PNG")&&!extension.equals(
```

Comment:
I don't think you need this. If the URL points towards a real image (whatever maybe the extension or even if none), if you have the img object as not null then you have a valid image. 
The purpose of getting the "extension" was just to find the second argument of the write() function. 
Let us discuss this.